### ISSUE TITLE
Getting all RL output values in Unity/Python

### ISSUE THREAD

**Issue by Zilch123** [Unity-Technologies/ml-agents#2527]
Please forgive for unsuitable vocabulary.

Is it possible to get all the values of the nn output ? In unity/Python. Is there an unity/Python function to get all the outputs.
What mean as output is the each action which maximises the reward. I would also need other action values for which reward might be less or zero. 
![image](https://user-images.githubusercontent.com/16478371/64598294-1d9f8880-d3d5-11e9-901a-aa514b5480d4.png)

**Comment by Zilch123** [comment]
https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Python-API.md

https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Learning-Environment-Design-Agents.md

**Comment by chriselion** [comment]
Sorry for the delay, glad you were able to find the answers.

**Comment by Zilch123** [comment]
> Sorry for the delay, glad you were able to find the answers.

No Problem @chriselion, I've a solution of implementing an RL algorithm in GYM and use that to get all the output from a NN.
I was wondering if you can point me a better way, I mean directly getting them in Unity by tinkering the of output function or something?

**Comment by Zilch123** [comment]
Thanks a lot @chriselion, I wasn't expecting you to answer me. Thank you!
The above will be useful, if I know how to load the trained .nn file and run it in jupyter. But I am not able to find the relevant documents for it.
I want to get the raw output of a trained RL agents, just today I came to know that the inference engine is doing this job.
https://github.com/Unity-Technologies/ml-agents/tree/master/UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain
I will figure out by reading all these Scripts. Thank you again!

**Comment by chriselion** [comment]
The `.nn` file (and some other files) are produced here:
https://github.com/Unity-Technologies/ml-agents/blob/fba79899eba43442af00524acab8ee220d800087/ml-agents/mlagents/trainers/tf_policy.py#L213-L235

The .nn file is a Unity-specific format for the Barracuda inference engine. But the `frozen_graph_def.pb` is a more standard tensorflow format that you should be able to visualize with standard tools.

**Comment by Zilch123** [comment]
Found out a similar Thread [unexpected results from Barracuda with pre-trained model](https://github.com/Unity-Technologies/ml-agents/issues/2027)
Have to implement import the .nn file (of a game) in C# and implement something similar to [this](https://www.dropbox.com/s/i983x9fydjtm0yd/TestTensorflow.cs?dl=0).
The[ Barracuda documentation](https://github.com/Unity-Technologies/ml-agents/blob/master/UnitySDK/Assets/ML-Agents/Plugins/Barracuda.Core/Barracuda.md) was what I was looking for. 
Thank you, 
Zilch

**Comment by Zilch123** [comment]
Hi @chriselion & @mantasp , I was wondering if there is any more documentation on [how to use Barracuda Inference engine](https://github.com/Unity-Technologies/ml-agents/blob/master/UnitySDK/Assets/ML-Agents/Plugins/Barracuda.Core/Barracuda.md). I mean like what are the functions in Barracuda Inference engine and a small description. 
Thank  you 
Zilch
